### PR TITLE
Update list table to oxygen UI ListingTable.DataGrid.Card component

### PR DESCRIPTION
**Pull request by jeradrutnam**
This pull request refactors the `ApplicationsList` and `FlowsList` components to improve their UI and code structure by replacing the old menu-based actions with inline row actions, updating the table rendering to use the new `ListingTable` abstraction, and simplifying event handling. It also updates the corresponding tests to reflect these UI changes and removes obsolete test logic. The changes enhance usability, maintainability, and consistency across the application.

This will be addressing: https://github.com/asgardeo/thunder/issues/1507

UI and component refactoring:

* Update oxygen-ui to v0.8.0, which has the DataGrid listing table support
* Migrated the table layout to use `ListingTable.Provider`, `ListingTable.Container`, and `ListingTable.DataGrid`, replacing the previous `Box` and `DataGrid.DataGrid` usage. (`[frontend/apps/thunder-develop/src/features/applications/components/ApplicationsList.tsxL230-R191](diffhunk://#diff-d186c2c51f129e391e6149294082bf6bba3457be01dfa8b23aa5a425412a84acL230-R191)`)
* Hide flows which are not clickable in the flow listing

image

image

Testing updates:

* Updated required tests for the new refactored changes

These changes streamline the codebase, improve user experience, and ensure tests remain reliable and relevant.

## Summary by CodeRabbit

* **New Features**
  * Added Applications to navigation under Resources.

* **Improvements**
  * Rebuilt lists (Applications, Flows, Users, User‑Types) to a unified listing table with row-click navigation and hover-visible per-row edit/delete actions and confirmation dialogs.
  * Improved avatar/name cells, per-row styling, flow filtering to hide non-editable flows, and minor view styling for schema tables.
  * Updated dashboard icons and navigation organization.

* **Tests**
  * Updated mocks and tests for the new listing table interactions and edge cases.

* **Chores**
  * Upgraded UI libraries to the next minor release.

### CODE REVIEW

File: frontend/apps/thunder-develop/src/features/users/components/UsersList.tsx [hunk 1]
```diff
@@ -368,61 +310,42 @@ export default function UsersList(props: UsersListProps) {
...
+            disableRowSelectionOnClick
+            localeText={dataGridLocaleText}
+            sx={{
+              '& .MuiDataGrid-row': {
+                cursor: 'pointer',
+              },
+            }}
+          />
+        </ListingTable.Container>
+      </ListingTable.Provider>
```
[the remark this answers is not in the document]
Comment (jeradrutnam) [reply frontend/apps/thunder-develop/src/features/users/components/UsersList.tsx:347]: Fixed

File: frontend/apps/thunder-develop/src/features/user-types/components/__tests__/UserTypesList.test.tsx [hunk 2]
```diff
@@ -265,31 +285,20 @@ describe('UserTypesList', () => {
...
-    await user.click(actionButtons[0]);
-
-    await waitFor(() => {
-      expect(screen.getByText('View')).toBeInTheDocument();
-      expect(screen.getByText('Delete')).toBeInTheDocument();
-    });
+    // The component uses inline delete buttons instead of a menu
+    const deleteButtons = screen.getAllByRole('button', {name: /delete/i});
+    expect(deleteButtons.length).toBeGreaterThan(0);
   });
```
[the remark this answers is not in the document]
Comment (jeradrutnam) [reply frontend/apps/thunder-develop/src/features/user-types/components/__tests__/UserTypesList.test.tsx]: Fixed

File: frontend/apps/thunder-develop/src/features/users/components/__tests__/UsersList.test.tsx [hunk 3]
```diff
@@ -311,39 +317,27 @@ describe('UsersList', () => {
...
-    await user.click(actionButtons[0]);
-
-    await waitFor(() => {
-      expect(screen.getByText('View')).toBeInTheDocument();
-      expect(screen.getByText('Delete')).toBeInTheDocument();
-    });
+    // Actions are now inline buttons (Trash2), not a dropdown menu
+    const deleteButtons = screen.getAllByRole('button', {name: /delete/i});
+    expect(deleteButtons.length).toBeGreaterThan(0);
   });
```
[the remark this answers is not in the document]
Comment (jeradrutnam) [reply frontend/apps/thunder-develop/src/features/users/components/__tests__/UsersList.test.tsx]: Fixed

File: frontend/apps/thunder-develop/src/features/applications/components/__tests__/ApplicationsList.test.tsx [hunk 4]
```diff
@@ -268,15 +283,36 @@ describe('ApplicationsList', () => {
...
+
+    const rows = screen.getAllByRole('row');
+    if (rows[0]) {
+      await user.click(rows[0]);
+
+      await waitFor(() => {
+        expect(mockNavigate).toHaveBeenCalledWith('/applications/app-1');
+      });
+    }
+  });
```
[the remark this answers is not in the document]
Comment (jeradrutnam) [reply frontend/apps/thunder-develop/src/features/applications/components/__tests__/ApplicationsList.test.tsx:309]: Fixed

File: frontend/apps/thunder-develop/src/features/applications/components/__tests__/ApplicationsList.test.tsx [hunk 5]
```diff
@@ -313,19 +347,16 @@ describe('ApplicationsList', () => {
...
-    await user.keyboard('{Escape}');
+      await waitFor(() => {
+        expect(mockNavigate).toHaveBeenCalledWith('/applications/app-1');
+      });
 
-    await waitFor(() => {
-      expect(screen.queryByRole('menu')).not.toBeInTheDocument();
-    });
+      expect(screen.getByRole('grid')).toBeInTheDocument();
+    }
```
[the remark this answers is not in the document]
Comment (jeradrutnam) [reply frontend/apps/thunder-develop/src/features/applications/components/__tests__/ApplicationsList.test.tsx]: This is a different line, Fixed all the occurrences.

File: frontend/apps/thunder-develop/src/features/users/components/UsersList.tsx [hunk 6]
```diff
@@ -368,61 +310,41 @@ export default function UsersList(props: UsersListProps) {
...
-            <Trash2 size={16} />
-          </ListItemIcon>
-          <ListItemText sx={{color: 'error.main'}}>{t('common:actions.delete')}</ListItemText>
-        </MenuItem>
-      </Menu>
+      <ListingTable.Provider variant="data-grid-card" loading={isLoading}>
+        <ListingTable.Container disablePaper>
+          <ListingTable.DataGrid
+            rows={userData?.users}
+            columns={columns}
```
[the remark this answers is not in the document]
Comment (jeradrutnam) [reply frontend/apps/thunder-develop/src/features/users/components/UsersList.tsx:317]: Fixed

File: frontend/apps/thunder-develop/src/features/users/components/__tests__/UsersList.test.tsx [hunk 7]
```diff
@@ -1038,25 +1025,17 @@ describe('UsersList', () => {
...
-      expect(screen.getByText('View')).toBeInTheDocument();
-    });
-
-    // Menu should be displayed
-    const viewButton = screen.getByText('View');
-    expect(viewButton).toBeInTheDocument();
+    // Each user row has its own inline delete button
+    const deleteButtons = screen.getAllByRole('button', {name: /delete/i});
+    expect(deleteButtons.length).toBeGreaterThanOrEqual(2);
   });
```
[the remark this answers is not in the document]
Comment (jeradrutnam) [reply frontend/apps/thunder-develop/src/features/users/components/__tests__/UsersList.test.tsx:1039]: Addressed
